### DISCUSSION TITLE
vsxmake生成的工程无法调试

### DISCUSSION THREAD

**Discussion by Domain** [xmake-io/xmake#2560] [Q&A]
很奇怪的问题，我用vsxmake生成的工程调试时，程序会异常退出。不是每个程序都会，而且直接命令行启动，或者用xmake run启动都没有问题，只有在vs中调试启动会，有些时候不调试启动也会。而且退出的错误码也不一样，有些是收到Ctrl+C退出（0xc000013a），有些又提示DLL初始化失败（0xc0000142），各种神奇的问题。命令行启动或者xmake run又完全正常，有些时候vs中不调试启动可以正常运行。
现在没什么头绪，不知道有没有其他人遇到这些问题。

**Comment by waruqi** [comment]
得看你有没有 add_packages 依赖一些动态库包了，如果依赖了，xmake run 会自动帮你绑定每个包的 path 环境，确保正常加载运行。

但是 vsxmake 里面，丢给 vs debugger 加载，目前只能直接透传 target file path ，vs 没法加载依赖包的环境的，也许会有加载失败或者异常的问题。。一种就是全改成静态库依赖，一种就是把依赖的 dll copy 到当前运行目录。

**Reply by Domain** [reply]
没有任何 add_packages，如果是DLL初始化失败，我可以理解为路径不对，但是收到Ctrl+C就比较奇怪了，而且出现时机也是随机的，可能main函数还没执行，就退出了，也可能运行了一段时间才收到

**Reply by waruqi** [reply]
那就不知道了，你自己调下。
